src/main/scala/org/tessellation/schema/Hom.scala

```diff
@@ -177,28 +175,76 @@ trait Hom[+A, +B] extends Ω {}
-  algebra: AlgebraM[M, F, B],
-  coalgebra: CoalgebraM[M, F, C]
-) extends Topos[A, B] {
-  protected def hyloM(implicit input: A => C) = scheme.hyloM(algebra, coalgebra).apply(input(data))
+
+
+class Cell[M[_] : Monad, F[_] : Traverse, A, B, S](val data: A, val hylo: S => M[B], val convert: A => S) extends Topos[A, B] {
+  def run(): M[B] = hylo(convert(data))
 }
 
 object Cell {
-  //  def create(data, algebra, coalgebra)
-  //  implicit val arrowInstance: Arrow[Cell] = new Arrow[Cell] {
-  //    override def lift[A, B](f: A => B): Cell[A, B] = ???
-  //
-  //    override def first[A, B, C](fa: Cell[A, B]): Cell[(A, C), (B, C)] = ???
-  //
-  //    override def compose[A, B, C](f: Cell[B, C], g: Cell[A, B]): Cell[A, C] =
-  //      // A ---> B ---> C
-  //      f.run(g.run(g)).asInstanceOf[Cell[A, C]]
-  ////      Cell(g.transform).transform
-  //  }
+  case class NullTerminal() extends Ω
+
+  def unapply[M[_], F[_], A, B, S](cell: Cell[M, F, A, B, S]): Some[(A, S => M[B])] = Some(cell.data, cell.hylo)
+
+  implicit def toCell[M[_], F[_], A <: Cell[M, F, Ω, Either[CellError, Ω], Ω]](a: A): Cell[M, F, Ω, Either[CellError, Ω], Ω] = a.asInstanceOf[Cell[M, F, Ω, Either[CellError, Ω], Ω]]
+
+  implicit def cellMonoid[M[_] : Applicative, F[_] : Applicative](implicit M: Monad[M], F: Traverse[F]): Monoid[Cell[M, F, Ω, Either[CellError, Ω], Ω]] = {
+    new Monoid[Cell[M, F, Ω, Either[CellError, Ω], Ω]] {
+
+      override def empty: Cell[M, F, Ω, Either[CellError, Ω], Ω] = {
+        val algebra = AlgebraM[M, F, Either[CellError, Ω]] {
+          _ => M.pure(NullTerminal().asInstanceOf[Ω].asRight[CellError])
+        }
+
+        val coalgebra = CoalgebraM[M, F, Ω] {
+          _ => M.compose[F].pure(NullTerminal().asInstanceOf[Ω])
+        }
+
+        val hyloM = scheme.hyloM(algebra, coalgebra)
+
+        new Cell[M, F, Ω, Either[CellError, Ω], Ω](
+          NullTerminal(),
+          hyloM,
+          _ => NullTerminal()
+        ) {
+          override def run(): M[Either[CellError, Ω]] = M.pure(NullTerminal().asInstanceOf[Ω].asRight[CellError])
+        }
+      }
+
+
+      // TODO: A param should be Ω as well to make it possible to combine Cells with different A type
+      override def combine(
+                            x: Cell[M, F, Ω, Either[CellError, Ω], Ω],
+                            y: Cell[M, F, Ω, Either[CellError, Ω], Ω]
+                          ): Cell[M, F, Ω, Either[CellError, Ω], Ω] = (x, y) match {
+        case (Cell(NullTerminal(), _), yy) => yy
+        case (xx, Cell(NullTerminal(), _)) => xx
+        case (Cell(NullTerminal(), _), Cell(NullTerminal(), _)) => empty
+        case (cella@Cell(a, _), cellb@Cell(b, _)) => {
+
+          val input: ΩList = a :: b
+
+          val combinedHyloM: Ω => M[Either[CellError, Ω]] = (a: Ω) => {
+            for {
+              aOutput <- cella.run()
+              bOutput <- cellb.run()
+              combinedOutput = aOutput.flatMap(aΩ => bOutput.map(bΩ => aΩ :: bΩ))
```

Comment:
hmm. is it exactly the same `cella.run().flatMap(a => cellb.run()`?
I'm thinking about a fast-fail but for comp is a syntax sugar so it should also not compute b output when a fails, right?

Reply:
Yes, it should fast-fail and put the very first error as `Left`. Actually that is open question how to call both cells (one after another? parallel?) so that fast-fail may be deterministic or not :D